packages/common/refresh-utils.ts

```diff
@@ -14,20 +14,30 @@ window.$RefreshSig$ = () => (type) => type;`
 export const getPreambleCode = (base: string): string =>
   preambleCode.replace('__BASE__', base)
 
-export function addRefreshWrapper<M extends { mappings: string } | undefined>(
+export const avoidSourceMapOption = Symbol()
+
+export function addRefreshWrapper<M extends { mappings: string }>(
   code: string,
-  map: M | string,
+  map: M | string | undefined | typeof avoidSourceMapOption,
   pluginName: string,
   id: string,
-): { code: string; map: M | string } {
+): { code: string; map: M | undefined | string } {
   const hasRefresh = refreshContentRE.test(code)
   const onlyReactComp = !hasRefresh && reactCompRE.test(code)
-  if (!hasRefresh && !onlyReactComp) return { code, map }
+  const newMap =
+    map === avoidSourceMapOption
+      ? undefined
+      : typeof map === 'string'
+        ? (JSON.parse(map) as M)
+        : map
+  if (!hasRefresh && !onlyReactComp) return { code, map: newMap }
```

Comment:
Is this wanted to return newMap in the no-op case?

Reply:
Ah, my bad.